src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -65,30 +62,14 @@ public class AutoroutePartHandler : ContentPartHandler<AutoroutePart>
 
         public override async Task PublishedAsync(PublishContentContext context, AutoroutePart part)
         {
-            // Remove entry if part is disabled.
-            if (part.Disabled)
-            {
-                await _entries.RemoveEntryAsync(part.ContentItem.ContentItemId, part.Path);
-            }
-
-            // Add parent content item path, and children, only if parent has a valid path.
-            if (!String.IsNullOrWhiteSpace(part.Path) && !part.Disabled)
+            // Update entries from the index table after the session is committed.
+            if (!String.IsNullOrWhiteSpace(part.Path))
             {
-                var entriesToAdd = new List<AutorouteEntry>
-                {
-                    new AutorouteEntry(part.ContentItem.ContentItemId, part.Path)
-                };
-
-                if (part.RouteContainedItems)
-                {
-                    _contentManager ??= _serviceProvider.GetRequiredService<IContentManager>();
-
-                    var containedAspect = await _contentManager.PopulateAspectAsync<ContainedContentItemsAspect>(context.PublishingItem);
-
-                    await PopulateContainedContentItemRoutes(entriesToAdd, part.ContentItem.ContentItemId, containedAspect, context.PublishingItem.Content as JObject, part.Path, true);
```

Comment:
@jtkech 

So `PopulateContainedContentItemRoutes` is still used by Pattern generation, but no longer here on published, if I am reading correctly.

There is something inside the method about homeroutes, which from memory allows a contained item to still be set as the home route.

So might still have to run some of it, for no purpose other than that?

Reply:
@deanmarcussen thanks for the feedback

From memory too ;) as i also started the same kind of PR for localization entries having their own singularities, so i will focus on it again this night, in the meantime,

> So PopulateContainedContentItemRoutes is still used by Pattern generation, but no longer here on published, if I am reading correctly.

Yes but the `AutoroutePartIndex` table is up to date and then re-used from its current cursor to update the entries.

> There is something inside the method about homeroutes, which from memory allows a contained item to still be set as the home route.

From memory i kept the part related to the homeroute for the container, about still allowing a contained item to be set as the home route, can't right now but i will check it this night